### DISCUSSION TITLE
Deprecate i2cdev mutex — I2C is thread-safe in IDF 5+

### DISCUSSION THREAD

**Discussion by rashedtalukder** [UncleRus/esp-idf-lib#567] [Ideas]
It appears that i2c master commands is now thread-safe in the ESP-IDF (e.g. https://docs.espressif.com/projects/esp-idf/en/latest/esp32/api-reference/peripherals/i2c.html#_CPPv420i2c_master_cmd_begin10i2c_port_t16i2c_cmd_handle_t10TickType_t). Since multiple mutexes waste cycles and make debugging difficult, should the i2cdev library's mutex be deprecated when using ESP-IDF 5+?

**Comment by UncleRus** [comment]
Yes, port mutexes are finally added to ESP-IDF. However, device mutexes are still needed, though not everywhere.

I would be grateful for the relevant pull request.
